filter/patterns_storage.go

```diff
@@ -88,7 +88,10 @@ func (storage *PatternStorage) matchPatterns(metric *ParsedMetric) []string {
 	seriesByTagPatternIndex := storage.SeriesByTagPatternIndex.Load().(*SeriesByTagPatternIndex)
 
 	matchedPatterns := make([]string, 0)
-	matchedPatterns = append(matchedPatterns, patternIndex.MatchPatterns(metric.Name)...)
-	matchedPatterns = append(matchedPatterns, seriesByTagPatternIndex.MatchPatterns(metric.Name, metric.Labels)...)
+	if len(metric.Labels) == 0 {
+		matchedPatterns = append(matchedPatterns, patternIndex.MatchPatterns(metric.Name)...)
+	} else {
+		matchedPatterns = append(matchedPatterns, seriesByTagPatternIndex.MatchPatterns(metric.Name, metric.Labels)...)
+	}
```

Comment:
Perhaps it is worthwhile to load index in conditional branches as well since only either `patternIndex` or `seriesByTagPatternIndex` is used?

Reply:
Done